lib/src/sdk_env.dart

```diff
@@ -8,43 +8,43 @@ import 'dart:io';
 
 import 'package:cli_util/cli_util.dart' as cli;
 import 'package:path/path.dart' as p;
+import 'package:pub_semver/pub_semver.dart';
 
 import 'analysis_options.dart';
 import 'logging.dart';
+import 'sdk_info.dart';
 import 'utils.dart';
 
-class DartSdk {
-  final String _sdkDir;
-  final Map<String, String> _environment = {};
-  final String _dartCmd;
+class DartSdk implements DartSdkInfo {
+  final String sdkDir;
+  final Map<String, String> _environment;
   final String _dartAnalyzerCmd;
   final String _dartfmtCmd;
   final String _pubCmd;
-  String _version;
+  final String dateString;
+  final String platform;
+  final Version version;
 
-  DartSdk._(String sdkDir, Map<String, String> environment)
-      : _sdkDir = sdkDir,
-        _dartCmd = _join(sdkDir, 'bin', 'dart'),
-        _dartAnalyzerCmd = _join(sdkDir, 'bin', 'dartanalyzer'),
+  DartSdk._(this.sdkDir, this._environment, this.version, this.dateString,
+      this.platform)
+      : _dartAnalyzerCmd = _join(sdkDir, 'bin', 'dartanalyzer'),
         _dartfmtCmd = _join(sdkDir, 'bin', 'dartfmt'),
-        _pubCmd = _join(sdkDir, 'bin', 'pub') {
-    if (environment != null) {
-      _environment.addAll(environment);
-    }
-  }
+        _pubCmd = _join(sdkDir, 'bin', 'pub');
 
-  factory DartSdk({String sdkDir, Map<String, String> environment}) =>
-      new DartSdk._(sdkDir ?? cli.getSdkPath(), environment);
+  factory DartSdk({String sdkDir, Map<String, String> environment}) {
+    sdkDir ??= cli.getSdkPath();
+    environment ??= {};
 
-  String get sdkDir => _sdkDir;
+    var dartCmd = _join(sdkDir, 'bin', 'dart');
 
-  String get version {
```

Comment:
I'd rather have something along the lines of:

```
DartSdkInfo _version;
Future<DartSdkInfo> get version async {
  _version ??= await _getVersion();
  return _version;
}

Future<String> _getVersion() async {
  // Process.run(_dartCmd, ['--version'), environment: _environment);
}
```

That way the initialization is kept simple.

Reply:
hrm...okay. Will ponder...

Reply:
I'd almost rather the ctor of the class fail than have an async error later...we can discuss next week

Reply:
One drawback is: we have no control over`Process.runSync`'s timeout. If it were in an async block (either by doing the creation in an `static Future<DartSdk> newDartSdk()`  method or in a separate `Future init()`, we could set a timeout. Maybe the static method would be a better approach here?

Reply:
Hrm. Sure. I'm not too worried about running this command timing out. If it fails, it'll fail immediately...unless the target system does some crazy RPC when a command is not found.

I love having the instance locked down right away.

Maybe put in an async factor method (not ctor)